src/widgets/vitals/vitals-biometrics-form/vitals-biometrics-form.utils.ts

```diff
@@ -0,0 +1,24 @@
+import { ConceptMetaData } from "./use-vitalsigns";
+import { inRange, isNumber } from "lodash";
+
+export function calculateBMI(weight: number, height: number): number {
+  if (weight > 0 && height > 0) {
+    return Number((weight / (height / 100) ** 2).toFixed(1));
+  }
+}
+
+export const isInValidRange = (
+  conceptsMetaData: Array<ConceptMetaData>,
+  conceptUuid: string,
+  value: string | number
+) => {
+  if (value === undefined || value === "") return false;
+  const normalRange = conceptsMetaData.find(c => c.uuid === conceptUuid);
+  return isNumber(normalRange.hiNormal) && isNumber(normalRange.lowNormal)
+    ? !inRange(
+        Number(value),
+        Number(normalRange.lowNormal),
+        Number(normalRange.hiNormal)
+      )
+    : false;
+};
```

Comment:
I think this function is upside-down. It's called `isInValidRange`, and it returns `true` when the value is bad, and `false` when the value is good (or there's no applicable range).

```suggestion
export const isInValidRange = (
  conceptsMetaData: Array<ConceptMetaData>,
  conceptUuid: string,
  value: string | number
) => {
  if (value === undefined || value === "") return true;
  const concept = conceptsMetaData.find(c => c.uuid === conceptUuid);
  return isNumber(concept.hiNormal) && isNumber(concept.lowNormal)
    ? inRange(
        Number(value),
        Number(concept.lowNormal),
        Number(concept.hiNormal)
      )
    : true;
};
```

Reply:
```suggestion
export const isInNormalRange = (
  conceptsMetaData: Array<ConceptMetaData>,
  conceptUuid: string,
  value: string | number
) => {
  if (value === undefined || value === "") return true;
  const concept = conceptsMetaData.find(c => c.uuid === conceptUuid);
  return isNumber(concept.hiNormal) && isNumber(concept.lowNormal)
    ? inRange(
        Number(value),
        Number(concept.lowNormal),
        Number(concept.hiNormal)
      )
    : true;
};
```